lieer/local.py

```diff
@@ -159,13 +139,62 @@ def set_file_extension (self, t):
         print ("Failed creating test file with file extension: " + t + ", not set.")
         raise
 
+
+  class State:
+    # last historyid of last synchronized message, anything that has happened
+    # remotely after this needs to be synchronized. gmail may return a 404 error
+    # if the history records have been deleted, in which case we have to do a full
+    # sync.
+    last_historyId = 0
+
+    # this is the last modification id of the notmuch db when the previous push was completed.
+    lastmod = 0
+
+    def __init__ (self, state_f, config_f):
+      self.state_f = state_f
+
+      if os.path.exists (self.state_f):
+        with open (self.state_f, 'r') as fd:
+          self.json = json.load (fd)
+      elif os.path.exists (config_f):
+        with open (config_f, 'r') as fd:
+          self.json = json.load (fd)
```

Comment:
One thing I thought of here is if the user tries to delete the state file just to do a clean sync, then we will pull in the settings from the config file. Perhaps we should clear those values from the config file if they were loaded from there (on the next write to the state).. maybe I am overcomplicating things now.

Reply:
Probably a good idea. We could write the config file immediately if it contains one of the state keys.

Reply:
Sounds good. I think with that in place this should be pretty much good to go.

Reply:
The latest commit takes a slightly more conservative approach, where during state load, both `config` and `state` are written immediately if the following conditions apply:

- A state file doesn't exist
- A config file does exist
- The config file contains `last_historyId` or `lastmod` keys

Reply:
Looks good!